src/AudioToolbox/MusicSequence.cs

```diff
@@ -274,6 +288,39 @@ public double GetBeatsForSeconds (double seconds)
 				return beats;
 			return 0;
 		}
+
+		[DllImport (Constants.AudioToolboxLibrary)]
+		extern static /* OSStatus */ MusicPlayerStatus MusicSequenceSetUserCallback (/* MusicSequence */ IntPtr inSequence, MusicSequenceUserCallbackProxy inCallback, /* void * */ IntPtr inClientData);
+
+		public void SetUserCallback (MusicSequenceUserCallback callback)
+		{
+			lock (userCallbackHandles) {
+				if (userCallbackHandles.ContainsKey (handle))
+					userCallbackHandles[handle] = callback;
+				else
+					userCallbackHandles.Add (handle, callback);
+			}
+
+			MusicSequenceSetUserCallback (handle, userCallbackProxy, IntPtr.Zero);
+		}
+
+#if !MONOMAC
+		[MonoPInvokeCallback (typeof (MusicSequenceUserCallbackProxy))]
+#endif
+		static void UserCallbackProxy (IntPtr inClientData, IntPtr inSequence, IntPtr inTrack, double inEventTime, IntPtr inEventData, double inStartSliceBeat, double inEndSliceBeat)
+		{
+			MusicSequenceUserCallback userCallback;
+			lock (userCallbackHandles)
+				userCallbackHandles.TryGetValue (inSequence, out userCallback);
+
+			if (userCallback != null) {
+				var userEventData = new MusicEventUserData (inEventData);
+				var musicSequence = MusicSequence.Lookup (inSequence);
+				var musicTrack = new MusicTrack (musicSequence, inTrack, false);
+
+				userCallback (musicTrack, inEventTime, userEventData, inStartSliceBeat, inEndSliceBeat);
```

Comment:
@spouliot @rolfbjarne please take a look at [MIDIPlayer](https://github.com/xamarin/private-samples/tree/master/ios9/MIDIPlayer) sample.

To invoke callback user of this API have to set events to `MusicTrack` and provide timestamp(when this event should be triggered)[1](https://github.com/xamarin/private-samples/blob/master/ios9/MIDIPlayer/MIDIPlayer/MIDIManager.cs#L242).

Theoretically user can set multiple events using the same timestamp. But most likely user events will be used to update UI(for instance draw sound waves while music track is playing).

In sample mentioned above user events a triggered(and hence user callback invoked) 1 call / sec.